crates/data_components/src/dynamodb/arrow.rs

```
@@ -0,0 +1,1632 @@
+
+    for item in items {
+        append_item_to_builders(item, &projected_schema, &mut builders)?;
+    }
+
+    let arrays = finish_builders(builders, &projected_schema)?;
+
+    RecordBatch::try_new(projected_schema, arrays).map_err(|e| Error::ConversionError {
+        source: Box::new(e),
+    })
+}
+
+fn create_empty_array(data_type: &DataType) -> ArrayRef {
+    match data_type {
+        DataType::Boolean => Arc::new(BooleanBuilder::new().finish()),
+        DataType::Int64 => Arc::new(Int64Builder::new().finish()),
+        DataType::Float64 => Arc::new(Float64Builder::new().finish()),
+        DataType::Utf8 => Arc::new(StringBuilder::new().finish()),
+        DataType::Binary => Arc::new(BinaryBuilder::new().finish()),
+        DataType::Date32 => Arc::new(Date32Builder::new().finish()),
+        DataType::Timestamp(TimeUnit::Millisecond, _) => Arc::new(
+            TimestampMillisecondBuilder::new()
+                .with_timezone(Arc::from("UTC"))
+                .finish(),
+        ),
+        DataType::List(field) => match field.data_type() {
+            DataType::Int64 => {
+                let values_builder = Int64Builder::new();
+                Arc::new(ListBuilder::new(values_builder).finish())
+            }
+            DataType::Float64 => {
+                let values_builder = Float64Builder::new();
+                Arc::new(ListBuilder::new(values_builder).finish())
+            }
+            DataType::Binary => {
+                let values_builder = BinaryBuilder::new();
+                Arc::new(ListBuilder::new(values_builder).finish())
+            }
+            _ => {
+                let values_builder = StringBuilder::new();
+                Arc::new(ListBuilder::new(values_builder).finish())
+            }
+        },
+        DataType::Null => Arc::new(NullBuilder::new().finish()),
+        _ => {
+            // Fallback to string for unsupported types
```

Is this safe to do? Shouldn't we return an error?